### PR TITLE
chore!: Update golang.org/x/ dependencies in api/

### PR DESCRIPTION
**Pull request by codingllama**
Update crypto and net in api/, plus pull a few commits that just landed.

### CODE REVIEW

**Comment by codingllama** [comment]
Follow up from #16591.

File: api/go.mod [hunk 1]
```diff
@@ -19,8 +19,8 @@ require (
 	go.opentelemetry.io/otel/sdk v1.7.0
 	go.opentelemetry.io/otel/trace v1.7.0
 	go.opentelemetry.io/proto/otlp v0.16.0
-	golang.org/x/crypto v0.0.0-20220126234351-aa10faf2a1f8
-	golang.org/x/net v0.0.0-20220127200216-cd36cc0744dd
+	golang.org/x/crypto v0.0.0-20220919173607-35f4265a4bc0
```
Comment (r0mant) [inline api/go.mod:22]: Anything potentially problematic in the crypto/ssh diff between these versions?

Comment (codingllama) [reply api/go.mod:22]: Not that I could spot (link for reference: https://cs.opensource.google/go/x/crypto;bpv=1)

**Review by r0mant — APPROVED** [review]

**Comment by codingllama** [comment]
Friendly ping @EdwardDowling @smallinsky ?

**Review by smallinsky — APPROVED** [review]

**Pull request merged**